core/src/main/java/com/google/zxing/oned/STFReader.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2008 ZXing authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.zxing.oned;
+
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.DecodeHintType;
+import com.google.zxing.FormatException;
+import com.google.zxing.NotFoundException;
+import com.google.zxing.Result;
+import com.google.zxing.ResultPoint;
+import com.google.zxing.common.BitArray;
+
+import java.util.Map;
+
+/**
+ * <p>Implements decoding of the STF format, or Standard Two of Five.</p>
+ *
+ * <p>This Reader is a blatant copy-and-paste from the {@link ITFReader} with
+ * modifications to account for the differences:</p>
+ * <ul>
+ * <li>Start and End patterns contain 2-wide wide lines.</li>
+ * <li>The patterns contain both the black and white lines.</li>
+ * </ul>
+ *
+ * <p>The checksum is optional and is not applied by this Reader. The consumer of the decoded
+ * value will have to apply a checksum if required.</p>
+ *
+ * <p>No standard documentation is available, the start/stop and digit patterns
+ * were obtained from http://www.barcodeisland.com/2of5.phtml</p>
+ *
+ * @author kevin.osullivan@sita.aero, SITA Lab. for the {@link ITFReader}
+ * @author lachezar for the Non-Interleaved changes
+ */
+public final class STFReader extends OneDReader {
+
+  private static final float MAX_AVG_VARIANCE = 0.38f;
+  private static final float MAX_INDIVIDUAL_VARIANCE = 0.78f;
+
+  private static final int W = 3; // Pixel width of a wide line
+  private static final int S = 2; // Pixel width of a start/stop wide line
+  private static final int N = 1; // Pixel width of a narrow line
+
+  /** Valid ITF lengths. Anything longer than the largest value is also allowed. */
+  private static final int[] DEFAULT_ALLOWED_LENGTHS = { 6, 8, 10, 12, 14 };
+
+  // Stores the actual narrow line width of the image being decoded.
+  private int narrowLineWidth = -1;
```

Comment:
It seems like this can easily not be a member, but a local var / argument?

Is there any more opportunity to refactor similarities with the ITF reader to reduce duplication?

Also, what's the general use case for this format? I hadn't heard of it.

I'm weighing usefuleness vs downside of copying and pasting code